aten/src/ATen/native/cuda/ForeachTernaryOp.cu

```diff
@@ -0,0 +1,249 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+#include <ATen/native/cuda/MultiTensorApply.cuh>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/NativeFunctions.h>
+#else
+#include <ATen/ops/_foreach_lerp_native.h>
+
+#include <ATen/ops/empty_like_native.h>
+#endif
+
+namespace at { namespace native {
+
+template <typename scalar_t>
+inline C10_HOST_DEVICE scalar_t lerp_(const scalar_t start, const scalar_t end, const scalar_t weight) {
+  using opmath_t = at::opmath_type<scalar_t>;
+  opmath_t start_val = start;
+  opmath_t end_val = end;
+  opmath_t weight_val = weight;
+  // Conditional for better numeric. This has been discussed in
+  // https://github.com/pytorch/pytorch/pull/18871
+  return (std::abs(weight_val) < 0.5)
+      ? start_val + weight * (end_val - start_val)
+      : end_val - (end_val - start_val) * (opmath_t{1} - weight_val);
+}
+
+// Ref: [GPU Pro Tip: Lerp Faster in C++](https://developer.nvidia.com/blog/lerp-faster-cuda/)
+template<>
+inline C10_HOST_DEVICE float lerp_(const float start, const float end, const float weight) {
```

Comment:
Why does it have underscore suffix? As far as I can see, it's not "inplace" in any sense

Reply:
mimicking PowKernel.cu https://github.com/pytorch/pytorch/blob/233305a852e1cd7f319b15b5137074c9eac455f6/aten/src/ATen/native/cuda/PowKernel.cu#L23-L27